packages/@jsii/kernel/src/kernel.ts

```diff
@@ -1325,6 +1304,44 @@ export class Kernel {
     return property;
   }
 
+  /**
+   * Shared (non-public implementation) to as not to break API recording.
+   */
+  private _getBinScriptCommand(
+    req: api.GetScriptCommandRequest,
+  ): api.GetScriptCommandResponse {
+    const packageDir = this._getPackageDir(req.assembly);
+    if (fs.pathExistsSync(packageDir)) {
+      // module exists, verify version
+      const epkg = fs.readJsonSync(path.join(packageDir, 'package.json'));
+
+      if (!epkg.bin) {
+        throw new JsiiFault(
+          'There is no bin scripts defined for this package.',
+        );
+      }
+
+      const scriptPath = epkg.bin[req.script];
+
+      if (!epkg.bin) {
+        throw new JsiiFault(`Script with name ${req.script} was not defined.`);
+      }
+
+      return {
+        command: path.join(packageDir, scriptPath),
```

Comment:
On Windows, this may only work if the caller spawns that commend via a `shell`, which is a relatively easy way to inadvertently make your code not work on Windows...

There would be a `.cmd` file that might be preferable to use as an entry point when on Windows if present...

Reply:
I don't disagree. I was taking a "do no harm" approach with this PR by just carrying forward the existing logic, which would also work on Windows.

I don't have easy access to a Windows host to test and understand the auto-generated `.cmd` file mentioned here: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#bin

Is it essential that Windows support be included in the PR, since this isn't a new regression?

Reply:
> Is it essential that Windows support be included in the PR, since this isn't a new regression?

Well... no. I have this principle that I never demand contributors leave the place _cleaner_ than they found it 😅